saleor/graphql/product/mutations/products.py

```diff
@@ -1157,7 +1229,12 @@ def perform_mutation(cls, _root, info, image_id, variant_id):
             )
         except models.VariantImage.DoesNotExist:
             raise ValidationError(
-                {"image_id": "Image is not assigned to this variant."}
+                {
+                    "image_id": ValidationError(
+                        "Image is not assigned to this variant.",
+                        code=ProductErrorCode.NOT_PRODUCTS_IMAGE,
```

Comment:
```suggestion
                        code=ProductErrorCode.NOT_FOUND,
```

Reply:
@NyanKiyoshi are you sure it's the same? There is a different between non-existing image and existing image that not belongs to that product.